httpx/_config.py

```diff
@@ -288,29 +289,47 @@ class PoolLimits:
 
     **Parameters:**
 
-    * **soft_limit** - Allow the connection pool to maintain keep-alive connections
+    * **max_keepalive** - Allow the connection pool to maintain keep-alive connections
                        below this point.
-    * **hard_limit** - The maximum number of concurrent connections that may be
+    * **max_connections** - The maximum number of concurrent connections that may be
                        established.
     """
 
     def __init__(
-        self, *, soft_limit: int = None, hard_limit: int = None,
+        self,
+        *,
+        max_keepalive: int = None,
+        max_connections: int = None,
+        soft_limit: int = None,
+        hard_limit: int = None,
     ):
-        self.soft_limit = soft_limit
-        self.hard_limit = hard_limit
+        self.max_keepalive = max_keepalive
+        self.max_connections = max_connections
+        if soft_limit is not None:  # pragma: nocover
+            self.max_keepalive = soft_limit
+            warnings.warn(
+                "'soft_limit' is deprecated. Use 'max_keepalive' instead.",
+                DeprecationWarning,
+            )
+        if hard_limit is not None:  # pragma: nocover
+            self.max_connections = hard_limit
+            warnings.warn(
+                "'hard_limit' is deprecated. Use 'max_connections' instead.",
+                DeprecationWarning,
+            )
```

Comment:
You can use the new [`warn_deprecated` util function.](https://github.com/encode/httpx/blob/master/httpx/_utils.py#L406-L407)